WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,20 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event)
+    event.id.in?(self.events.map(&:id))
+  end
+
+  def belongs_to_region?(region_or_country)
+    (self.countryId == region_or_country) || (self.country.continentId == region_or_country)
+  end
+
+  def search(search_param)
+    if search_param != nil
+      (name.include? search_param) || (cityName.include? search_param) || (venue.include? search_param)
```

Comment:
What about case sensitivity? What about searches like https://www.worldcubeassociation.org/results/competitions.php?eventId=&regionId=&years=only%2B2014&pattern=duss&list=List (I searched for "duss", but it found "Düsseldorf").

When you fix this, please add a test for this as well.